src/appleseed/renderer/modeling/light/sunlight.cpp

```diff
@@ -412,12 +414,17 @@ namespace
             }
         }
 
-        static double compute_sun_radius(const double distance)
+        static double compute_sun_radius(const double distance, const double scene_diameter)
         {
-            // angular diameter = sun diameter / distance -> sun diameter = angular diameter * distance
-            // sun diameter = 0.009301 radians * distance
-            // sun radius = 0.0046505 radians * distance
-            return 0.0046505 * distance;
+            // sun diameter = 1.3914
+            // angular_diameter = 2 * arctan(sun_diameter / (2 * distance))
+            // angular_diameter / 2 = arctan(sun_diameter / (2 * distance))
+            // tan(angular_diameter / 2) * distance = sun_radius
+            // tan(angular_diameter / 2) * scene_diameter = virtual_sun_radius
+
+            float angular_diameter = 2 * atan(1.3914 / (2 * distance));
+
+            return tan(angular_diameter / 2 ) * scene_diameter;
```

Comment:
And, it turns out that this expression simplifies itself to:

    return 1.3914 / (2 * distance) * scene_diameter

Maybe what you ended up with earlier?

Reply:
I'd suggest making 1.3914 a constant, e.g. `SunDiameter`.

Reply:
then how about make 0.6957 `SunRadius`, rather than `1.3914 / 2`?

Reply:
Yep!